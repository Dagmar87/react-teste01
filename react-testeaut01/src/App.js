import React, { Component } from "react";
import PropTypes from "prop-types";
import Home from "./Home";

const fakeDatabase = {
  Dagmar: [
    "8h00 - Assistir as aulas de Matematica na escola",
    "12h30 - Encontrar Renata para o almoço",
    "15h00 - Encontrar Fabio no Clube de Futebol",
    "20h00 - Jantar à beira da piscina com Pedro e Lorena",
  ],
  Inaiana: [
    "8h00 - Reunião com o CREA",
    "15h00 - Encontrar Gabriel no Clube de Arquitetura",
  ],
};

class App extends Component {
  static propTypes = {
    isLoggedIn: PropTypes.bool,
    username: PropTypes.string,
  };

  render() {
    const { isLoggedIn, username } = this.props;

    return (
      <div className="App">
        {
          isLoggedIn
            ? <Home username={username} tasks={fakeDatabase[username]} />
            : <p>Olá, visitante. Faça login para continuar.</p>
        }
      </div>      
    );
  }
}

export default App;
