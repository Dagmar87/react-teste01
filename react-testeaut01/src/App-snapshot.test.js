import React from 'react';
import App from './App';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

it("deve renderizar a página inicial para o usuario não logado", () => {
    const props = {
        isLoggedIn: false,
        username: null
    }

    const app = mount(<App {...props} />)

    expect(toJson(app)).toMatchSnapshot()
})

it("deve renderizar a página inicial para o usuario logado (Dagmar)", () => {
    const props = {
        isLoggedIn: true,
        username: "Dagmar"
    }

    const app = mount(<App {...props} />)

    expect(toJson(app)).toMatchSnapshot()
})

it("deve renderizar a página inicial para o usuario logado sem tarefas", () => {
    const props = {
        isLoggedIn: true,
        username: "Divaldo"
    }

    const app = mount(<App {...props} />)

    expect(toJson(app)).toMatchSnapshot()
})