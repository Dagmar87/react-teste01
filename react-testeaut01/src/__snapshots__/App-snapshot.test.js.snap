// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deve renderizar a página inicial para o usuario logado (Dagmar) 1`] = `
<App
  isLoggedIn={true}
  username="Dagmar"
>
  <div
    className="App"
  >
    <Home
      tasks={
        Array [
          "8h00 - Assistir as aulas de Matematica na escola",
          "12h30 - Encontrar Renata para o almoço",
          "15h00 - Encontrar Fabio no Clube de Futebol",
          "20h00 - Jantar à beira da piscina com Pedro e Lorena",
        ]
      }
      username="Dagmar"
    >
      <div>
        <p>
          Bem-vindo, 
          Dagmar
          !
        </p>
        <ul>
          <li
            key="0"
          >
            8h00 - Assistir as aulas de Matematica na escola
          </li>
          <li
            key="1"
          >
            12h30 - Encontrar Renata para o almoço
          </li>
          <li
            key="2"
          >
            15h00 - Encontrar Fabio no Clube de Futebol
          </li>
          <li
            key="3"
          >
            20h00 - Jantar à beira da piscina com Pedro e Lorena
          </li>
        </ul>
      </div>
    </Home>
  </div>
</App>
`;

exports[`deve renderizar a página inicial para o usuario logado sem tarefas 1`] = `
<App
  isLoggedIn={true}
  username="Divaldo"
>
  <div
    className="App"
  >
    <Home
      username="Divaldo"
    >
      <div>
        <p>
          Bem-vindo, 
          Divaldo
          !
        </p>
        <ul />
      </div>
    </Home>
  </div>
</App>
`;

exports[`deve renderizar a página inicial para o usuario não logado 1`] = `
<App
  isLoggedIn={false}
  username={null}
>
  <div
    className="App"
  >
    <p>
      Olá, visitante. Faça login para continuar.
    </p>
  </div>
</App>
`;
